File: internal/comment/model/comment.go
package model

type Comment struct {
	ID       string   `json:"id"`
	BeerID   string   `json:"beerId"`
	UserID   string   `json:"userId"`
	Text     string   `json:"text" validate:"required"`
	Likes    int      `json:"likes"`
	LikedBy  []string `json:"likedBy"`
	Positive bool     `json:"positive" validate:"required"`
	Created  string   `json:"created"`
}

File: lib/features/beer/domain/entities/comment.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'comment.freezed.dart';

@freezed
class Comment with _$Comment {
  const factory Comment({
    required int id,
    required int beerId,
    required int userId,
    required String text,
    required int likesCount,
    List<int>? likedBy,
    String? userName,
    DateTime? createdAt,
    @Default(false) bool isBlocked,
    @Default(false) bool needsReview,
    bool? positive,
  }) = _Comment;
}
